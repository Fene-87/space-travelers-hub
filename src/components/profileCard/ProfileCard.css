.profile-section {
  display: flex;
  flex-direction: column;
  margin: 4rem 0 0 50px;
}

.profile-section > h2 {
  font-family: 'lato', sans-serif;
}

.name-mission {
  display: flex;
  flex-direction: column;
  border: 1px solid #d3d3d3;
  border-radius: 5px;
  border-bottom: none;
  margin-top: 10px;
}

.row {
  border-bottom: 2px solid #d3d3d3;
  padding: 20px 15px;
}

.row p {
  font-size: 18px;
}

.row .text-muted {
  font-size: 25px;
  color: red;
  font-weight: bolder;
  text-align: center;
  margin-top: 25px;
}
