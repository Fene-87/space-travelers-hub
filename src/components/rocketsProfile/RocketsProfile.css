.rockets-container {
  margin-top: 4rem;
}

.rockets-header {
  margin-top: 1rem;
}

.rockets-list {
  margin-top: 0.5rem;
  border: 1px solid #d3d3d3;
}

.rockets-list > li {
  padding: 10px 20px;
}
