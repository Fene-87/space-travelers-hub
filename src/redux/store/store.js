import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from '../rockets/rocketsSlice';
import missionReducer from '../missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
    missions: missionReducer,
  },
});

export default store;
