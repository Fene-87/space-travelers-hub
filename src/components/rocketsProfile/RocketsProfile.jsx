import React from 'react';
import { useSelector } from 'react-redux';
import './RocketsProfile.css';

const RocketsProfile = () => {
  const { rocketList } = useSelector((store) => store.rockets);
  const reservedRockets = rocketList.filter((rocket) => {
    return rocket.reserved;
  });

  return (
    <div className="rockets-container">
      <h2 className="rockets-header">My Rockets</h2>
      <ul className="rockets-list">
        {reservedRockets.map((rocket) => (
          <li>{rocket.name}</li>
        ))}
      </ul>
    </div>
  );
};

export default RocketsProfile;
