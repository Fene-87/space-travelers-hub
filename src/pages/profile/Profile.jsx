import React from 'react';
import { useSelector } from 'react-redux';
import Navbar from '../../components/navbar/Navbar';
import { selectAllMissions } from '../../redux/missions/missionsSlice';
import RocketsProfile from '../../components/rocketsProfile/RocketsProfile';
import './profile.css';

const Profile = () => {
  const missions = useSelector(selectAllMissions);
  const joinedMissions = missions.filter((mission) => mission.reserved === true);
  if (joinedMissions.length === 0) {
    return (
      <div>
        <Navbar />
        <div className="row">
          <p className="text-muted">No joined missions</p>
        </div>
      </div>
    );
  }
  return (
    <div>
      <Navbar />
      <section className="profile-container">
        <div className="profile-section">
          <h2>My Missions</h2>
          <div className="name-mission">
            {joinedMissions.map((mission) => (
              <div className="row" key={mission.id}>
                <p>{mission.missionName}</p>
              </div>
            ))}
          </div>
        </div>
        <RocketsProfile />
      </section>
    </div>
  );
};

export default Profile;
