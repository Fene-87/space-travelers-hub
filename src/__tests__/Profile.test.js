import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store/store';
import Profile from '../pages/profile/Profile';
import ProfileCard from '../components/profileCard/ProfileCard';
import RocketsProfile from '../components/rocketsProfile/RocketsProfile';

describe('testing snapshot with Profile', () => {
  it('testing Profile component', () => {
    const profile = render(
      <Provider store={store}>
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </Provider>,
    );
    expect(profile).toMatchSnapshot();
  });
  it('testing ProfileCard component', () => {
    const profileCard = render(
      <Provider store={store}>
        <ProfileCard />
      </Provider>,
    );
    expect(profileCard).toMatchSnapshot();
  });
  it('testing RocketsProfile component', () => {
    const rocketsProfile = render(
      <Provider store={store}>
        <RocketsProfile />
      </Provider>,
    );
    expect(rocketsProfile).toMatchSnapshot();
  });
});
