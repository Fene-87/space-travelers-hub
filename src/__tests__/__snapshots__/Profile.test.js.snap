// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing snapshot with Profile testing Profile component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <section
          class="navbar"
        >
          <div
            class="left-side"
          >
            <img
              alt="logo"
              class="logo-img"
              src="planet.png"
            />
            <h2
              class="nav-header"
            >
              Space Travelers' Hub
            </h2>
          </div>
          <div
            class="right-side"
          >
            <ul
              class="nav-links"
            >
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  Rockets
                </a>
              </li>
              <li
                class="boundary"
              >
                <a
                  class=""
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="/profile"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </div>
        </section>
        <div
          class="profile-container"
        >
          <div
            class="card"
          >
            <div>
              <div
                class="row"
              >
                <p
                  class="text-muted"
                >
                  No joined missions
                </p>
              </div>
            </div>
          </div>
          <div
            class="card"
          >
            <div>
              <div
                class="row"
              >
                <p
                  class="text-muted"
                >
                  No Reserved Rockets
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <section
        class="navbar"
      >
        <div
          class="left-side"
        >
          <img
            alt="logo"
            class="logo-img"
            src="planet.png"
          />
          <h2
            class="nav-header"
          >
            Space Travelers' Hub
          </h2>
        </div>
        <div
          class="right-side"
        >
          <ul
            class="nav-links"
          >
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li
              class="boundary"
            >
              <a
                class=""
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class=""
                href="/profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </section>
      <div
        class="profile-container"
      >
        <div
          class="card"
        >
          <div>
            <div
              class="row"
            >
              <p
                class="text-muted"
              >
                No joined missions
              </p>
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div>
            <div
              class="row"
            >
              <p
                class="text-muted"
              >
                No Reserved Rockets
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshot with Profile testing ProfileCard component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="row"
        >
          <p
            class="text-muted"
          >
            No joined missions
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="row"
      >
        <p
          class="text-muted"
        >
          No joined missions
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshot with Profile testing RocketsProfile component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="row"
        >
          <p
            class="text-muted"
          >
            No Reserved Rockets
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="row"
      >
        <p
          class="text-muted"
        >
          No Reserved Rockets
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
